metaserver/src/main/scala/scala/meta/languageserver/Effects.scala

```diff
@@ -0,0 +1,18 @@
+package scala.meta.languageserver
+
+/**
+ * The ScalametaLanguageServer effects.
+ *
+ * Observable[Unit] is not descriptive of what the observable represents.
+ * Instead, we create Unit-like types to better document what effects are
+ * flowing through our application.
```

Comment:
Love this.

Reply:
Poor man's type-safety. I refactored to this in fact because I hit on bugs related to accidentally creating `Observable[Observable[T]]` from a `.map` when I should have been using `.flatMap` to build `Observable[Unit]`

Reply:
are these phantom types?